### PR TITLE
[finalize-warmboot.sh] reset cpufreq governor to default

### PR DESCRIPTION
**Pull request by stepanblyschak**
#### Why I did it

Set cpufreq.default_governor to *performance* for faster boot time. We observe consistent 1 sec improvement across several devices.

The change in finalize-warmboot.sh restores the default governor after fast or warm boot is finished.

**NOTE**: This will apply to upgrades starting from 202405 since this is set in shutdown path to avoid any extra scripts running at boot time. Upgrade from older versions/branches will require a runtime patch to fast-reboot and warm-reboot script.

##### Work item tracking
- Microsoft ADO **(number only)**:

#### How I did it

After fast or warm boot is finished restore to default governor.

#### How to verify it

Run fast-reboot or warm-reboot. Check:
```
admin@sonic:~$ cat /sys/devices/system/cpu/cpu0/cpufreq/scaling_governor
performance
```

After boot is finalized check that it is reset back to default:
```
admin@sonic:~$ cat /sys/devices/system/cpu/cpu0/cpufreq/scaling_governor
schedutil
```

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211
- [ ] 202305
- [ ] 202311
- [x] 202405

#### Tested branch (Please provide the tested image version)

- [ ] 
- [ ]

### CODE REVIEW

**Review by volodymyrsamotiy — APPROVED** [review]

File: files/image_config/warmboot-finalizer/finalize-warmboot.sh [hunk 1]
```diff
@@ -101,16 +101,30 @@ function check_list()
     echo ${RET_LIST}
 }
 
+function set_cpufreq_governor() {
+    governor="$1"
+    echo "$governor" | tee /sys/devices/system/cpu/cpu*/cpufreq/scaling_governor \
```
Comment (nazariig) [inline files/image_config/warmboot-finalizer/finalize-warmboot.sh:106]: @stepanblyschak tee `stdout` is redundant. Can we hide it?

**Review by nazariig — CHANGES_REQUESTED** [review]

File: files/image_config/warmboot-finalizer/finalize-warmboot.sh [hunk 2]
```diff
@@ -101,16 +101,30 @@ function check_list()
     echo ${RET_LIST}
 }
 
+function set_cpufreq_governor() {
+    governor="$1"
```
Comment (nazariig) [inline files/image_config/warmboot-finalizer/finalize-warmboot.sh:105]: @stepanblyschak `local -r` ?

**Review by nazariig — CHANGES_REQUESTED** [review]

**Review by nazariig — CHANGES_REQUESTED** [review]

File: files/image_config/warmboot-finalizer/finalize-warmboot.sh (hunk 1 above)
Comment (stepanblyschak) [reply files/image_config/warmboot-finalizer/finalize-warmboot.sh]: @nazariig  done

File: files/image_config/warmboot-finalizer/finalize-warmboot.sh (hunk 2 above)
Comment (stepanblyschak) [reply files/image_config/warmboot-finalizer/finalize-warmboot.sh]: @nazariig  done

File: files/image_config/warmboot-finalizer/finalize-warmboot.sh [hunk 3]
```diff
@@ -101,16 +101,30 @@ function check_list()
     echo ${RET_LIST}
 }
 
+function set_cpufreq_governor() {
```
[the remark this answers is not in the document]
Comment (stepanblyschak) [reply files/image_config/warmboot-finalizer/finalize-warmboot.sh:104]: @nazariig  don't see a benefit from such check

**Review by nazariig — APPROVED** [review]

**Pull request closed without merging**